### PR TITLE
task(accounts): Create sms.manager

### PR DESCRIPTION
**Pull request by dschom**
## Because
- We want a SMS manager to send messages

## This pull request
- Creates an sms manager that wraps sending of twilio messages
- Ensures that manager is injectable with NestJS
- Creates a twilio provider
- Ensures that provider is injectable with NestJS
- Introduces the twilio as a root level package dependency

## Issue that this pull request solves

Closes: FXA-10336

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

### CODE REVIEW

File: libs/accounts/recovery-phone/src/lib/twilio.provider.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/twilio.provider.ts:1]: We need this at some point, but I didn't see a task for it in the backlog. It probably shouldn't go in `lib/accounts/recovery-phone`. Looking for feedback where to put it. Or perhaps we can just file a followup if this needs to be more robust.

File: libs/accounts/recovery-phone/src/lib/sms.manager.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+@Injectable()
+export class SmsManager {
+  constructor(
+    @Inject(TwilioProvider) private readonly client: Twilio,
+    private readonly config: SmsManagerConfig
+  ) {}
+
+  public async sendSMS({ to, body }: { to: string; body: string }) {
+    return await this.client.messages.create({
```
Comment (dschom) [inline libs/accounts/recovery-phone/src/lib/sms.manager.ts:18]: Does this need to be more complex? 
Do we want to put checks on to or body? 
Do we need to check and handle the result?

**Review by vbudhram — APPROVED** [review]
@dschom I like it, on a side note, I wonder how we will do our integration tests with this? Perhaps a testing interface that sends and email instead?

File: libs/accounts/recovery-phone/src/lib/twilio.provider.ts (hunk 1 above)
Comment (vbudhram) [reply libs/accounts/recovery-phone/src/lib/twilio.provider.ts:1]: Seems fine to update in follow up.

**Comment by dschom** [comment]
> @dschom I like it, on a side note, I wonder how we will do our integration tests with this? Perhaps a testing interface that sends and email instead?

That's a good question. A lot of times I'll forgo integration tests on libs, since the API's that uses them will typically have integrations and that's where the configuration is actually configured (or misconfigured). I think your point still stands. At some point we will want to make sure it actually can send a sms. I was also wondering how we'd want to go about this, and I am not really sure. I have similar questions about functional testing for sms flows. Let's talk about it at our next weekly.

File: libs/accounts/recovery-phone/src/lib/twilio.provider.ts (hunk 1 above)
Comment (bbangert) [reply libs/accounts/recovery-phone/src/lib/twilio.provider.ts:1]: Makes sense for this to live here until we actually need it outside of recovery phone functionality.

**Comment by dschom** [comment]
@vbudhram Added the checks and tests for them. Thanks for the feedback.

**Pull request merged**